app/scripts/lib/setupAccountLabelsPetnamesBridge.ts

```diff
@@ -0,0 +1,105 @@
+import { NameController, NameType } from '@metamask/name-controller';
+import { cloneDeep } from 'lodash';
+import log from 'loglevel';
+
+export type IdentityEntry = {
+  address: string;
+  name: string;
+};
+
+export type PertinentState = {
+  identities: { [address: string]: IdentityEntry };
+};
+
+export type PertinentPreferencesController = {
+  store: {
+    getState: () => PertinentState;
+    subscribe: (callback: (state: PertinentState) => void) => void;
+  };
+};
+
+const MAINNET_CHAIN_ID = '0x1';
```

Comment:
This highlights an important consideration.

Petnames are stored in the `NameController` with unique names and proposed names per variation. In the case of ethereum address petnames, these variations are the chain IDs. This means we currently need to create the new petname entries for every applicable chain. To achieve this, we can either:

1. Update the listener to still fire on the preferences controller state change but also whenever the network changes, at which point we can create the relevant petnames to match the account identities for the current chain. The downside of this is we grow and duplicate the state alot over time if the user interacts with many networks, plus it adds complexity to this bridge logic.

2. Use a special variation for these petnames such as `*` and update the petnames UI components to check that variation as a fallback if it can't find a petname for a specific chain ID. This means we only have to define these petnames once so the state is smaller, plus no listening to network changes, plus a much simpler migration as we only have to worry about this generic variation (`*`) rather than every relevant network.

In both of these solutions, at least for now, the petnames UI will contain to only override the entry for the current chain, rather than the generic entry. Perhaps in future we could let users choose to save a new petname for all chains or a single one.

Reply:
I've added FALLBACK_VARIATION to the useName hook. PTAL.